lib/enkf/enkf_main.cpp

```diff
@@ -1107,31 +1104,97 @@ static void enkf_main_analysis_update( enkf_main_type * enkf_main ,
 
-        if (analysis_module_check_option( module , ANALYSIS_UPDATE_A)){
-          if (analysis_module_check_option( module , ANALYSIS_ITERABLE)){
-            analysis_module_updateA( module , localA , S , R , dObs , E , D , module_info, enkf_main->shared_rng);
-          }
-          else
-            analysis_module_updateA( module , localA , S , R , dObs , E , D , module_info, enkf_main->shared_rng);
-        } else {
-          if (analysis_module_check_option( module , ANALYSIS_USE_A)){
-            analysis_module_initX( module , X , localA , S , R , dObs , E , D, enkf_main->shared_rng);
+        /*
+          The update for one local_dataset instance consists of two main chunks:
+
+          1. The first chunk updates all the parameters which don't have row
+             scaling attached. These parameters are serialized together to the A
+             matrix and all the parameters are updated in one go.
+
+          2. The second chunk is loop over all the parameters which have row
+             scaling attached. These parameters are updated one at a time.
+         */
+
+
+        // Part 1: Parameters which do not have row scaling attached.
+        {
+          enkf_main_serialize_dataset(enkf_main_get_ensemble_config(enkf_main), dataset , step2 ,  use_count , tp , serialize_info);
+
+          if (analysis_module_check_option( module , ANALYSIS_UPDATE_A)){
+            if (analysis_module_check_option( module , ANALYSIS_ITERABLE)){
+              analysis_module_updateA( module , localA , S , R , dObs , E , D , module_info, enkf_main->shared_rng);
+            }
+            else
+              analysis_module_updateA( module , localA , S , R , dObs , E , D , module_info, enkf_main->shared_rng);
+          } else {
+            if (analysis_module_check_option( module , ANALYSIS_USE_A)){
+              analysis_module_initX( module , X , localA , S , R , dObs , E , D, enkf_main->shared_rng);
+            }
+
+            matrix_inplace_matmul_mt2( A , X , tp );
           }
 
-          matrix_inplace_matmul_mt2( A , X , tp );
+          enkf_main_deserialize_dataset( enkf_main_get_ensemble_config( enkf_main ) , dataset , serialize_info , tp);
         }
 
-        // The enkf_main_deserialize_dataset() function will dismantle the A
-        // matrix from the serialize_info structure and distribute that content
-        // over to enkf_node instances and eventually the storage layer.
-        enkf_main_deserialize_dataset( enkf_main_get_ensemble_config( enkf_main ) , dataset , serialize_info , tp);
 
+        // Part 2: Parameters with row scaling attached - to support distance based localization.
+        {
+          stringlist_type * scaled_keys = local_dataset_alloc_scaled_keys(dataset);
+          if (stringlist_get_size(scaled_keys)) {
+            throw std::logic_error("Sorry - the use of row scaling for distance based localization is not yet implemented");
```

Comment:
are these `throws` necessary?

Reply:
Well - I consider a kind of feature toggling; something has gone way south if this exception is ever thrown. I would prefer to let it remain as a marker?

Reply:
Let us keep it for now, but I would prefer to have testable code coming in to libres. In this PR the tests are a little lacking, so lets aim to fix this in a future ;)